citesphere/src/test/java/edu/asu/diging/citesphere/core/util/model/impl/CitationHelperTest.java

```diff
@@ -0,0 +1,376 @@
+            while (p1.hasNext() && p2.hasNext()) {
+                if (!comparePerson(p1.next().getPerson(), p2.next().getPerson())) {
+                    return false;
+                }
+            }
+            return true;
+
+        }
+
+        return false;
+
+    }
+
+    private boolean compareTags(Set<ICitationConceptTag> a1, Set<ICitationConceptTag> a2) {
+
+        if (a1 != null && a2 != null) {
+
+            if (a1.size() != a2.size())
+                return false;
+            Iterator<ICitationConceptTag> p1 = a1.iterator();
+            Iterator<ICitationConceptTag> p2 = a2.iterator();
+            while (p1.hasNext() && p2.hasNext()) {
+                if (!compareConceptTag(p1.next(), p2.next())) {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        return false;
+
+    }
+
+    private boolean equalsCitation(ICitation c1, ICitation c2) {
+
+        if (!c1.getTitle().equals(c2.getTitle())) {
+            return false;
+        }
+
+        if (!c1.getShortTitle().equals(c2.getShortTitle())) {
+            return false;
+        }
+
+        if (!comparePersons(c1.getAuthors(), c2.getAuthors())) {
+            return false;
+        }
+
+        if (!comparePersons(c1.getEditors(), c2.getEditors())) {
+            return false;
+        }
+
+        if (!compareCreators(c1.getOtherCreators(), c2.getOtherCreators())) {
+            return false;
+        }
+
+        if (!compareTags(c1.getConceptTags(), c2.getConceptTags())) {
+            return false;
+        }
+
+        return true;
+
+    }
+
+    private boolean comparePerson(IPerson p1, IPerson p2) {
+
+        if (!p1.getFirstName().equals(p2.getFirstName()) || !p1.getLastName().equals(p2.getLastName())) {
+            return false;
+        }
+        if (p1.getAffiliations() != null && p2.getAffiliations() != null) {
+            if (p1.getAffiliations().size() != p2.getAffiliations().size())
+                return false;
+            Iterator<IAffiliation> i1 = p1.getAffiliations().iterator();
+            Iterator<IAffiliation> i2 = p2.getAffiliations().iterator();
+            while (i1.hasNext() && i2.hasNext()) {
+                if (!compareAffliattion(i1.next(), i2.next())) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private boolean compareAffliattion(IAffiliation a1, IAffiliation a2) {
+
+        if (a1.getName().equals(a2.getName())) {
+            return true;
+        }
+
+        return false;
+
+    }
+
+    private boolean compareConceptTag(ICitationConceptTag c1, ICitationConceptTag c2) {
+
+        if (c1.getLocalConceptId().equals(c2.getLocalConceptId())
+                && c1.getLocalConceptTypeId().equals(c2.getLocalConceptTypeId())) {
+            return true;
+        }
+
+        return false;
+    }
+
+    class ConceptArgMatcher extends ArgumentMatcher<ICitationConcept> {
```

Comment:
these arguemtn matchers shoudl be initialized with the concept/type, otherwise if we add more testcases that need to return different concepts, we can't reuse the argument matchers.

Reply:
Do I need to create separate classes for argument matchers?

Reply:
not sure what you mean by that? You already have separate classes, don't you?

Reply:
Separate classes instead of inner classes so they can be used by other tests

Reply:
we assume new tests for this class; if it turns out other classes can use the matchers, we will move them into separate files then.

Reply:
oh ok, 
I didn't understand "argument matchers should be initialized with concept/type".